Modules/FIT/FDD/src/OutOfBunchCollFeeModulesCheck.cxx

```diff
@@ -0,0 +1,156 @@
+        int bin = 0;
+        float value = 0;
+        try {
+          bin = std::stoi(metainfo.first);
+          value = std::stof(metainfo.second);
+          allCollPerFeeModule[bin] = value;
+        } catch (const std::invalid_argument& e) {
+          ILOG(Warning, Support) << "Could not get value for key " << metainfo.first << ENDM;
+          continue;
+        }
+      }
+
+      // Calculate out-of-bunch-coll fraction for Fee Modules
+      for (int binY = 1; binY <= histogram->GetNbinsY(); binY++) {
+        auto outOfBcCollisions = histogram->Integral(1, sBCperOrbit, binY, binY);
+        float fraction = 0;
+        if (allCollPerFeeModule[binY]) {
+          fraction = outOfBcCollisions / allCollPerFeeModule[binY];
+        }
+
+        if (fraction > mFractionOutOfBunchColl) {
+          mFractionOutOfBunchColl = fraction;
+        }
+      }
+
+      // Check the biggest fraction of out-of-bunch-coll
+      if (mFractionOutOfBunchColl > mThreshError) {
+        result.set(Quality::Bad);
+        result.addReason(FlagReasonFactory::Unknown(),
+                         Form("Fraction of out of bunch collisions (%.2e) is above \"Error\" threshold (%.2e)",
+                              mFractionOutOfBunchColl, mThreshError));
+      } else if (mFractionOutOfBunchColl > mThreshWarning) {
+        result.set(Quality::Medium);
+        result.addReason(FlagReasonFactory::Unknown(),
+                         Form("Fraction of out of bunch collisions (%.2e) is above \"Warning\" threshold (%.2e)",
+                              mFractionOutOfBunchColl, mThreshWarning));
+      } else {
+        result.set(Quality::Good);
+      }
+    }
+  }
+
+  return result;
+}
+
+std::string OutOfBunchCollFeeModulesCheck::getAcceptedType() { return "TH2"; }
+
+void OutOfBunchCollFeeModulesCheck::beautify(std::shared_ptr<MonitorObject> mo, Quality checkResult)
+{
+  if (mo->getName().find("OutOfBunchColl_BCvsFeeModules") != std::string::npos) {
+    auto* histogram = dynamic_cast<TH2*>(mo->getObject());
+
+    if (!histogram) {
+      ILOG(Error, Support) << "beautify(): MO " << mo->getName() << " not found" << ENDM;
+      return;
+    }
+
+    TPaveText* msg = new TPaveText(0.1, 0.85, 0.9, 0.9, "NDC");
+    histogram->GetListOfFunctions()->Add(msg);
+    msg->SetName(Form("%s_msg", mo->GetName()));
+    msg->Clear();
+    if (checkResult.isWorseThan(Quality::Good)) {
+      msg->AddText(checkResult.getReasons()[0].second.c_str());
+    }
+    int color = kWhite;
+    if (checkResult == Quality::Good) {
+      color = kGreen + 1;
+      msg->AddText(">> Quality::Good <<");
+    } else if (checkResult == Quality::Medium) {
+      color = kOrange - 1;
+      msg->AddText(">> Quality::Medium <<");
+    } else if (checkResult == Quality::Bad) {
+      color = kRed;
+      msg->AddText(">> Quality::Bad <<");
+    }
+    msg->SetFillColor(color);
+  }
+}
+
+} // namespace o2::quality_control_modules::fdd
```

Comment:
Does it have same logic as https://github.com/AliceO2Group/QualityControl/blob/master/Modules/FIT/FIT/src/LevelCheck.cxx ? You can calculated histogram out_of_coll/in_coll (with fee module per x-axis) and then just apply LevelCheck task